tests/test_equilibrium.py

```diff
@@ -464,3 +466,156 @@ def test_backward_compatible_load_and_resolve():
     f_obj = ForceBalance(eq=eq)
     obj = ObjectiveFunction(f_obj)
     eq.solve(maxiter=1, objective=obj)
+
+
+def test_shifted_circle_geometry():
+    """
+    In this test, we calculate a low-beta shifted circle equilibrium with DESC.
+
+    We then compare the various geometric coefficients with their respective analytical
+    expressions.
+    """
+    R0 = 1
+    a0 = 0.1
+
+    surf = FourierRZToroidalSurface(
+        R_lmn=[R0, a0],
+        Z_lmn=[0.0, -a0],
+        modes_R=np.array([[0, 0], [1, 0]]),
+        modes_Z=np.array([[0, 0], [-1, 0]]),
+        sym=True,
+        NFP=1.0,
+    )
+
+    pressure = PowerSeriesProfile([1.5e5, 0, -1.5e5])
+    iota = PowerSeriesProfile([0.5, 0, -0.2])
+
+    eq = Equilibrium(
+        L=10, M=10, N=0, surface=surf, Psi=0.1, pressure=pressure, iota=iota
+    )
+
+    eq, _ = eq.solve(objective="force", ftol=1e-3, gtol=1e-6, maxiter=30, verbose=3)
+
+    eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+    psi = 0.25  # rho^2 (or normalized psi)
+    alpha = 0
+
+    eq_keys = ["iota", "iota_r", "a", "rho", "psi"]
+
+    data_eq = eq.compute(eq_keys)
+
+    iotas = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota"])
+    shears = np.interp(np.sqrt(psi), data_eq["rho"], data_eq["iota_r"])
+
+    N = int((2 * eq.M_grid) * 4 + 1)
+
+    zeta = np.linspace(-1.0 * np.pi / iotas, 1.0 * np.pi / iotas, N)
+    theta_PEST = alpha * np.ones(N, dtype=int) + iotas * zeta
+
+    coords1 = np.zeros((N, 3))
+    coords1[:, 0] = np.sqrt(psi) * np.ones(N, dtype=int)
+    coords1[:, 1] = theta_PEST
+    coords1[:, 2] = zeta
+
+    # Creating a grid along a field line
+    c1 = eq.compute_theta_coords(coords1)
+    grid = Grid(c1, sort=False)
+
+    data_keys = [
+        "kappa",
+        "|grad(psi)|^2",
+        "grad(|B|)",
+        "grad(alpha)",
+        "grad(psi)",
+        "B",
+        "grad(|B|)",
+        "iota",
+        "|B|",
+        "B^zeta",
+        "cvdrift0",
+        "cvdrift",
+        "gbdrift",
+    ]
+
+    data = eq.compute(data_keys, grid=grid, override_grid=False)
+
+    psib = data_eq["psi"][-1]
+
+    # signs
+    sign_psi = psib / np.abs(psib)
+    sign_iota = iotas / np.abs(iotas)
+
+    # normalizations
+    Lref = data_eq["a"]
+    Bref = 2 * np.abs(psib) / Lref**2
+
+    modB = data["|B|"]
+    bmag = modB / Bref
+
+    x = Lref * np.sqrt(psi)
+    shat = -x / iotas * shears / Lref
+
+    grad_psi = data["grad(psi)"]
+    grad_alpha = data["grad(alpha)"]
+
+    iota = data["iota"]
+
+    gradpar = Lref * data["B^zeta"] / modB
+
+    gds21 = -sign_iota * np.array(dot(grad_psi, grad_alpha)) * shat / Bref
```

Comment:
might be worth adding compute functions for these as well (or aliases to existing ones)

Reply:
I think the only one missing is gds2 = |grad(alpha)|^2 but that is in the ballooning branch.

Reply:
This test is only needed so that Kaya can use these formulae for bounce averaging. A much more thorough tests for the coefficients is in the adjoint ballooning branch.

Reply:
`s_hat` instead of `shat`?